client-http/src/main/java/com/cloudera/livy/client/http/LivyConnection.java

```diff
@@ -90,24 +91,40 @@ public boolean isContentCompressionEnabled() {
       }
     };
 
-    Credentials dummyCredentials = new Credentials() {
-      @Override
-      public String getPassword() {
-        return null;
+    Credentials credentials;
+    // If user info is specified in the url, pass them to the CredentialsProvider.
+    if (uri.getUserInfo() != null) {
+      String[] userInfo = uri.getUserInfo().split(":");
+      if (userInfo.length != 2) {
+        throw new IllegalArgumentException("Malformed user info in the url.");
       }
-
-      @Override
-      public Principal getUserPrincipal() {
-        return null;
+      try {
+        String username = URLDecoder.decode(userInfo[0], "UTF-8");
```

Comment:
Minor: is it better to use `UTF_8.name` instead of string. In Spark such usage of string all changed to `UTF_8` object.